endpoints/openrtb2/auction_test.go

```diff
@@ -4650,6 +4654,69 @@ func TestValidateStoredResp(t *testing.T) {
 	}
 }
 
+func TestValidResponseWhenRequestRejected(t *testing.T) {
+	nbr := openrtb3.NoBidReason(123)
+	reject := hookexecution.RejectError{
+		int(nbr),
+		hookexecution.HookID{ModuleCode: "foobar", HookCode: "foo"},
+		hooks.StageEntrypoint.String(),
+	}
+
+	testCases := []struct {
+		description         string
+		expectedBidResponse openrtb2.BidResponse
+		hookExecutor        hookexecution.HookStageExecutor
+	}{
+		{
+			"Assert correct BidResponse when request rejected at entrypoint stage",
+			openrtb2.BidResponse{ID: "some-request-id", NBR: &nbr},
+			rejectableHookExecutor{entrypointReject: &reject},
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.description, func(t *testing.T) {
+			actualAuctionObject := analytics.AuctionObject{}
+			logger := newMockLogger(nil, &actualAuctionObject)
+			deps := &endpointDeps{
+				fakeUUIDGenerator{},
+				&nobidExchange{},
+				mockBidderParamValidator{},
+				&mockStoredReqFetcher{},
+				empty_fetcher.EmptyFetcher{},
+				empty_fetcher.EmptyFetcher{},
+				&config.Configuration{MaxRequestSize: maxSize},
+				&metricsConfig.NilMetricsEngine{},
+				logger,
+				map[string]string{},
+				false,
+				[]byte{},
+				openrtb_ext.BuildBidderMap(),
+				nil,
+				nil,
+				hardcodedResponseIPValidator{response: true},
+				empty_fetcher.EmptyFetcher{},
+				test.hookExecutor,
+			}
+
+			reqBody := validRequest(t, "../exemplary/simple.json")
+			req := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(reqBody))
+			recorder := httptest.NewRecorder()
+
+			deps.Auction(recorder, req, nil)
+			assert.Equal(t, recorder.Code, http.StatusOK, "Endpoint should return 200 OK.")
```

Comment:
According to the spec, a rejection in the entrypoint stage should result in a 204 or 4XX:

> For /openrtb2/auction and /openrtb2/video modules must be able to modify the original POST request that comes into the server or reject the request entirely. Rejection at this stage may result in HTTP 204 or a 4xx error and a metric associating the rejection with the module.

In that case, the body should be empty if using 204 or contain an error message if using 4XX.

What is PBS-Java doing?

Reply:
Yes, we've read this but PBS-Java is always sending a 200 response with an empty bid response, so we decided to do the same for PBS-Go